src/components/editor/MonacoEditor.tsx

```diff
@@ -72,27 +78,48 @@ function MonacoEditor({
     //     // eslint-disable-next-line react-hooks/exhaustive-deps
     // }, [serverUpdate]);
 
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-    const editorUpdateDebounce = useCallback(
-        debounce((val: string | undefined) => {
-            if (val && onChange) {
-                setIsChanging(true);
-                onChange(JSON.parse(val), path, currentCatalog.spec_type)
+    const updateValue = () => {
+        const currentValue = editorRef.current?.getValue();
+        if (onChange && currentValue) {
+            setStatus(EditorStatus.EDITING);
+
+            let parsedVal;
+            try {
+                parsedVal = JSON.parse(currentValue);
+            } catch {
+                setStatus(EditorStatus.INVALID);
+            }
+
+            if (parsedVal) {
+                setStatus(EditorStatus.SAVING);
+                onChange(
+                    parsedVal,
+                    currentCatalog.catalog_name,
+                    currentCatalog.spec_type
+                )
                     .then(() => {
-                        setIsChanging(false);
+                        setStatus(EditorStatus.SAVED);
                     })
                     .catch(() => {
-                        setIsChanging(false);
+                        setStatus(EditorStatus.SAVE_FAILED);
                     });
+            } else {
+                setStatus(EditorStatus.INVALID);
             }
-        }, 750),
-        []
-    );
+        }
+    };
+
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    const debouncedChange = useCallback(debounce(updateValue, 1000), [
+        currentCatalog,
+    ]);
```

Comment:
Create a debounce function and only recreate when the currentCataog changes.

Reply:
1 second lag might be too long... we'll need to figure that out. 500 was too tight and 750 felt good... but we'll need to see feedback from folks.